ui/src/lib/findings.ts

```diff
@@ -0,0 +1,105 @@
+/* Copyright 2022 Corsha.
+   Licensed under the Apache License, Version 2.0 (the 'License');
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an 'AS IS' BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+/**
+ * Analysis describes the results of an analysis type
+ **/
+export interface Analysis {
+  /** A machine-friendly ID identifying the CAST analysis, e.g. C-104-reused-auth */
+  id: string;
+
+  /** Human-readable name of analysis */
+  name: string;
+
+  /** Human-readable description of what the analysis is */
+  description: string;
+
+  /** A number used for sorting analyses in the UI, sorted ascending */
+  priority: number;
+
+  /** An RFC3339-format timestamp of when this analysis most recently completed executing; i.e. its last-updated time */
+  lastUpdated: string;
+
+  /** The result of the analysis */
+  findings: Finding[];
+}
+
+/** The set of severity levels */
+export const Severity = ["none", "low", "medium", "high", "critical"] as const;
+
+/** The severity of a finding. */
+export type Severity = typeof Severity[number];
+
+/** When a finding occurred. It might be a time span or at an instant */
+export type OccurredAt =
+  | {
+      /** The start date and time formatted as a UTC RFC-3339 string */
+      start: string;
+
+      /** The end date and time formatted as a UTC RFC-3339 string */
+      end;
+      string;
```

Comment:
should probably be like this (?)
```suggestion
      /** The end date and time formatted as a UTC RFC-3339 string */
      end: string;
```

Reply:
fixed